### ISSUE TITLE
Block detected on certain videos

### ISSUE THREAD

**Issue by ivanst02** [futo-org/grayjay-android#1662]
### What happened?

On certain videos, constantly getting:
Using IOS sources fallback (cached) -> Block detected attempting fallback -> **Media Error** The media source encountered an unauthorized error...

Tried reinstalling the plugin, but it did not help. Fidgeted with the plugin options also, no change (Youtube activity, logout, etc.).

Android 14

### Grayjay Version

278

### What plugins are you seeing the problem on?

Youtube

### Plugin Version

219

### When do you experience the issue?

- [X] While logged in
- [X] While logged out
- [ ] N/A

### Are you using a VPN?

No

### Relevant log output

51085fbe-7532-4086-b823-978cb47a3a23

**Comment by abster95** [comment]
I have similar issues. I noticed that when I'm on wifi it immediately triggers the "block detected", but on mobile data I can usually get the forst chunk of the video to play and then it triggers block detected around 1 minute and enters reset loop

**Comment by max-ishere** [comment]
Getting this, but differently. The first video might load fine but everything after that gets a block detected attempting bypass error on the 1:00 minute mark for all videos, even the ones that played just fine. It fixes itself after some time but breaks after a couple videos. I can't tell how long it takes for it to work again, but its not something that fixes itself on app restart.

**Comment by Zvonimir-FUTO** [comment]
Related to #1610 and #1596

**Comment by Epzillon** [comment]
Adding some more information here. This ONLY happens when I am connected to a VPN, regardless of VPN settings or GrayJay settings. I have also tried activating the "Force Experimental UMP Streams" setting, but that seems to cause other issues no matter whether I use the VPN or not (still unable to play YouTube videos that is).

Curiously I reside in Sweden and use Mullvad VPN (based in Sweden) and am connected to a server which is also in Sweden. I have tried enabling and disabling certain VPN settings (Multihop, DAITA [with and without Direct Hop] and DNS content blockers) but none seems to make a difference.

Simply being connected to a VPN seems to cause videos to not play for more than 59 seconds.

**Comment by jonasvn77** [comment]
I experience the same issue with videos stopping exactly at 1.00 mins after I have watched the first. Today I managed to watch some kind of live stream for several hours, so I thought it was ok again... But now same problem occurs- cant connect to channel or something, is the error message. 
Seems like yourtube are able to cut the connection in a new way that Grayjay cant handle... 

I know I don't sound especially competent but Ive used Grayjay several hours daily for a long period, but never earlier have I experienced a problem I couldn't fix only using logic, basic knowledge and tips like switching on/off ' and of course reading relevant info provided  when  installing directly from the source (not Play)...
But now I have tried everything ... 
Seems

**Comment by sithi3** [comment]
This issue of certain videos won't play seems to be either a problem of Youtube blocking these videos, or a problem of different programming code contents / structure between the official Youtube app (YouTube ReVanced, athough modified, is also kind of the official app) and non-official YouTube apps such as Grayjay, NewPipe. The supporting reason is that I've found a few videos that simply wouldn't play at all (i.e. blank screen) in Grayjay (message saying Block detected, attempting bypass) and in NewPipe (message saying An error occurred, see the notification). These were exactly the same videos (same video id). In contrast, when I tried to play these same videos in either the Official YouTube or YouTube Revanced, they load and play perfectly.

Notes: 
1) NewPipe notifications reported such errors as: Network error happened. Player error[tppe=ERROR_CODE_IO_BAD_HTTP_STATUS] occurred while playing https://www.youtube.com/watch?v=xxxxx  (xxxxx represents the video ID of the video in question). Additionally, it also gave details of the errors encountered by exoplayer2 (package name=com.google.android.exoplayer2)
2) This blocking issue might occur very recently. A couple days ago, I made a playlist of 10 songs in NewPipe. Every song was perfectly playable. Then the next day, one song in the list got this problem

I'd also like to take this opportunity to report my finding that while YouTube, Youtube ReVanced, and NewPipe gave virtually identical search results; but in Grayjay, it was quite noticeable that few videos could be missing from the search results.

My Grayjay version is 278, YouTube plugin version is 219. Both are the same as those of the original poster. My Android version is Android 14, One UI 6.1.1. The issue occurred while logged out. As for while logged out, I did not try (I tried to stay away from Google as much as possible. On this Android device, I use only Aurora Store, and did not add Google account / log in on this device)

In summary, this issue of certain video blocking might occur very recently (i.e. started from very few days ago), and could probably also happen with any apps not based on the official app. However, the official YouTube app, and the apps based on it (e.g. YouTube Revanced) should be free from this problem.

Best regards

**Comment by pearcebasmanm** [comment]
This occurs for me also (when my vpn is on), but I thought it worth noting that the behavior does not occur on the new Grayjay Desktop application, even with the same vpn enabled. Perhaps the desktop app is doing something different, which would be worth looking into.

**Comment by Epzillon** [comment]
That is interesting, for Grayjay Desktop I can still only play 1 minute with my VPN Connected. Interestingly I could not use Split Tunneling to accept the GrayJay executable to bypass the VPN, this just made anything refuse to load, even thumbnails. Unsure why that happens. I do believe my initial comment should have maybe been put on #1610 instead however as this seems more relevant.

**Comment by JordyScript** [comment]
On desktop I get the following error

![image](https://github.com/user-attachments/assets/7049c1b4-a642-4466-88fc-02440dd16d44)

Reloading doesn't fix the problem, because it still stops the video from loading beyond the first minute. 

On mobile I get

> Using login fallback to resolve : Sign in to prove you are not a bot

and 

> Failed to get iOS stream using UMP fallback instead

After that I cannot watch for more than one minute again.

All of these findings are the same both with and without mullvad VPN on

**Comment by max-ishere** [comment]
Got this crash, this time it was on the 47th second

```
Version information (version_name = 278, version_code = 278, flavor = stable, build_type = release)
Device information (brand= google, manufacturer = Google, device = oriole, version-sdk = 35, version-os = )

(e, ExceptionActivity, 2025-01-04 22:54:35): Uncaught exception ("Unknown Context"): [Youtube] Failed to request page [408] (414)[3-4]

com.futo.platformplayer.engine.exceptions.ScriptException: [Youtube] Failed to request page [408] (414)[3-4]
	at com.futo.platformplayer.constructs.BatchedTaskHandler$execute$1$task$1.invokeSuspend(BatchedTaskHandler.kt:62)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:108)
	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:584)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:793)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:697)
	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:684)
	Suppressed: kotlinx.coroutines.internal.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@308892, Dispatchers.IO]
Caused by: com.caoccao.javet.exceptions.JavetExecutionException: Error: Failed to request page [408]
	at com.caoccao.javet.interop.V8Native.scriptExecute(Native Method)
	at com.caoccao.javet.interop.V8Runtime.execute(V8Runtime.java:928)
	at com.caoccao.javet.interop.executors.V8StringExecutor.execute(V8StringExecutor.java:107)
	at com.caoccao.javet.interop.IV8Executable.execute(IV8Executable.java:44)
	at com.futo.platformplayer.engine.V8Plugin$executeTyped$2.invoke(V8Plugin.kt:220)
	at com.futo.platformplayer.engine.V8Plugin$executeTyped$2.invoke(V8Plugin.kt:219)
	at com.futo.platformplayer.engine.V8Plugin$Companion.catchScriptErrors(V8Plugin.kt:294)
	at com.futo.platformplayer.engine.V8Plugin.catchScriptErrors(V8Plugin.kt:269)
	at com.futo.platformplayer.engine.V8Plugin.executeTyped(V8Plugin.kt:219)
	at com.futo.platformplayer.api.media.platforms.js.JSClient$getContentDetails$1.invoke(JSClient.kt:520)
	at com.futo.platformplayer.api.media.platforms.js.JSClient$getContentDetails$1.invoke(JSClient.kt:517)
	at com.futo.platformplayer.api.media.platforms.js.JSClient.isBusyWith(JSClient.kt:733)
	at com.futo.platformplayer.api.media.platforms.js.JSClient.getContentDetails(JSClient.kt:517)
	at com.futo.platformplayer.states.StatePlatform$_batchTaskGetVideoDetails$1.invokeSuspend(StatePlatform.kt:117)
	at com.futo.platformplayer.states.StatePlatform$_batchTaskGetVideoDetails$1.invoke(Unknown Source:8)
	at com.futo.platformplayer.states.StatePlatform$_batchTaskGetVideoDetails$1.invoke(Unknown Source:4)
	at com.futo.platformplayer.constructs.BatchedTaskHandler$execute$1$task$1.invokeSuspend(BatchedTaskHandler.kt:55)
	... 6 more
```

**Comment by Epzillon** [comment]
JordyScript, I get the same thing on GrayJay Desktop, I opened an issue on the Desktop repo: https://github.com/futo-org/Grayjay.Desktop/issues/258

**Comment by JustCryen** [comment]
Is there a way to switch sources for the specific channel in cases like this? 
So, for example, when I know Louis has an Odysee channel (which Is linked), can I watch his videos there until the YouTube ones will be fixed?

I can't find a way to select Odysee version of the videos on his channel unless I subscribe to his Odysee channel as well.
Which I believe is not how GrayJay is supposed to work.

**Comment by Zvonimir-FUTO** [comment]
No, unfortunately not yet.

**Comment by ayoahha** [comment]
Same issue here apparently
"Failed to get iOS stream data, fallback to UMP" when I am on 4G, and strangely when I launch a video that is not my language!

On WiFi, it is OK, those videos are playing.

**Comment by Zvonimir-FUTO** [comment]
Hi everyone, this has been finally fixed and new version of Grayjay and Youtube have been released!

Go update and test and tell us if you find issues except the one below.

We are aware that age restricted, controversial videos, or other videos that require login may not yet work. We will be working on this next week.

The desktop release that is required for this fix will also be released next week.
